metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1341,7 +1357,7 @@ class MetalsLanguageServer(
     ) {
       event.eventType match {
         case EventType.Create =>
-          buildTargets.onCreate(path)
+          mainBuildTargetsData.onCreate(path)
```

Comment:
Could we instead of differentiating the different connections, just find targetData this path might belong to?

Reply:
We should probably do something like this, yeah.

Reply:
Fixed.